api/src/controllers/schema.ts

```diff
@@ -0,0 +1,120 @@
+import { parseJSON } from '@directus/shared/utils';
+import Busboy from 'busboy';
+import express, { RequestHandler } from 'express';
+import { load as loadYaml } from 'js-yaml';
+import { InvalidPayloadException, UnsupportedMediaTypeException } from '../exceptions';
+import logger from '../logger';
+import { respond } from '../middleware/respond';
+import { SchemaService } from '../services/schema';
+import { Snapshot } from '../types';
+import asyncHandler from '../utils/async-handler';
+import { getVersionedHash } from '../utils/get-versioned-hash';
+
+const router = express.Router();
+
+router.get(
+	'/snapshot',
+	asyncHandler(async (req, res, next) => {
+		const service = new SchemaService({ accountability: req.accountability });
+		const currentSnapshot = await service.snapshot();
+		res.locals.payload = { data: currentSnapshot };
+		return next();
+	}),
+	respond
+);
+
+router.post(
+	'/apply',
+	asyncHandler(async (req, _res, next) => {
+		const service = new SchemaService({ accountability: req.accountability });
+		await service.apply(req.body);
+		return next();
+	}),
+	respond
+);
+
+const schemaMultipartHandler: RequestHandler = (req, res, next) => {
+	if (req.is('application/json')) {
+		if (Object.keys(req.body).length === 0) throw new InvalidPayloadException(`No data were included in the body`);
+		res.locals.uploadedSnapshot = req.body;
+		return next();
+	}
+
+	if (!req.is('multipart/form-data')) throw new UnsupportedMediaTypeException(`Unsupported Content-Type header`);
+
+	const headers = req.headers['content-type']
+		? req.headers
+		: {
+				...req.headers,
+				'content-type': 'application/octet-stream',
+		  };
+
+	const busboy = Busboy({ headers });
```

Comment:
Lets add `limits: { fileSize }` here with a env var for max schema size to process. Related to ENG-491

Reply:
Alternatively we could hardcode it to a safe upper limit (like 5MB), but this feels like something that should be configurable through env

Reply:
Actually lets combine this with ENG-491 and just make it a single config option for both 👍🏻